### PR TITLE
Feat/optimizations

### PR DESCRIPTION
**Pull request by jonty007**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,77 +1,183 @@
...
+  useInterval(async () => {
+    try {
+      if (computer) {
+        const newBalance = await computer.db.wallet.getBalance();
+        const newRevs = await computer.getRevs(
+          computer.db.wallet.getPublicKey()
+        );
+        const newArts = await Promise.all(
+          newRevs.map(async (rev) => computer.sync(rev))
+        );
```
Comment (ClemensLey) [inline src/App.js]: Sync is a potentially slow operation as it needs to do a recursive analysis of the blockchain. We want to minimize the number of calls to "sync".

One way to do it is to initialize two arrays "revs" and "artworks" to empty arrays. Then in a loop:
* use getRevs to query for the latest revs (call that array something line newRevs)
* then compute "deleted revs" and "new revs" by comparing the newly computed revs to latest revs
* remove the artworks in "deleted revs" from artworks
* use sync to compute the new artworks and add them to the "artworks" array

File: src/App.js [hunk 2]
```diff
@@ -1,77 +1,183 @@
...
+  const [artworks, setArtworks] = useState([]);
+  // const [refresh, setRefresh] = useState(0);
 
-  useEffect(() => {
-    const fetchArtworks = async () => {
-      setArtworks(await Promise.all(revs.map(async rev => computer.sync(rev))))
-    }
-    fetchArtworks()
-  }, [revs, computer])
+  function useInterval(callback, delay) {
```
Comment (ClemensLey) [inline src/App.js:33]: This is a better way to implement polling

**Review by ClemensLey — COMMENTED** [review]
Cool

**Comment by ltardivo** [comment]
Looks good. Merging to master.

**Pull request merged**